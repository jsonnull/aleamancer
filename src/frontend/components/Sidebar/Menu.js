// @flow
import React from 'react'
import styled from 'styled-components'
import type { Tab } from 'common/types'
import MenuItem from './MenuItem'

const MenuContainer = styled.div`
  display: flex;
  flex-direction: row;
  height: 4.8rem;
  color: black;
  background-color: ${props =>
    props.theme.name === 'dark' ? props.theme.backgroundSecondary : ''};
`

type MenuButton = [Tab, string] | 'separator'

const buttons: Array<MenuButton> = [
  ['Session', 'fa-globe'],
  ['Character', 'fa-id-card-o']
]

type Props = {
  tab: Tab,
  changeTab: Function
}
class Menu extends React.Component<Props> {
  render() {
    return (
      <MenuContainer>
        {buttons.map(button => {
          const [name, icon] = button
          const selected = name == this.props.tab

          return (
            <MenuItem
              key={name}
              name={name}
              icon={icon}
              selected={selected}
              action={this.props.changeTab}
            />
          )
        })}
      </MenuContainer>
    )
  }
}

export default Menu
