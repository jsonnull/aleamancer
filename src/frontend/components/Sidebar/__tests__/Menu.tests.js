// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import Menu from '../Menu.js'

describe('Sidebar Menu component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Menu tab="Session" changeTab={() => {}} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
