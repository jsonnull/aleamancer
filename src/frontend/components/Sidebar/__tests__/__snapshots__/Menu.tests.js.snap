// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Menu component renders correctly 1`] = `
.c1 {
  text-align: center;
  height: 4.8rem;
  width: 4.8rem;
  line-height: 4.8rem;
  background-repeat: no-repeat;
  background-size: 100% 4px;
  background-position: 0 4.4rem;
  background-image: linear-gradient(black 0%,black 100%);
  -webkit-transition: background-size 100ms,background-position 100ms;
  transition: background-size 100ms,background-position 100ms;
}

.c1:hover {
  background-color: #057091;
  color: white;
  cursor: pointer;
}

.c2 {
  text-align: center;
  height: 4.8rem;
  width: 4.8rem;
  line-height: 4.8rem;
  background-repeat: no-repeat;
  background-size: 0 4px;
  background-position: 50% 4.4rem;
  background-image: linear-gradient(black 0%,black 100%);
  -webkit-transition: background-size 100ms,background-position 100ms;
  transition: background-size 100ms,background-position 100ms;
}

.c2:hover {
  background-color: #057091;
  color: white;
  cursor: pointer;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 4.8rem;
  color: black;
}

<div
  className="c0"
>
  <div
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div>
      <div
        className="c1"
        onClick={[Function]}
        selected={true}
      >
        <i
          className="fa fa-globe"
        />
      </div>
    </div>
  </div>
  <div
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div>
      <div
        className="c2"
        onClick={[Function]}
        selected={false}
      >
        <i
          className="fa fa-id-card-o"
        />
      </div>
    </div>
  </div>
</div>
`;
